recipe/tests/tzdb.cpp

```diff
@@ -88,6 +92,19 @@ test_load_leapseconds()
 
 int main()
 {
+  // test implementation of finding tzdb
+  std::string tz_dir = __gnu_cxx::zoneinfo_file("");
+  std::printf("using tzdb-path: %s\n", tz_dir.c_str());
+  const char* conda_prefix = getenv("CONDA_PREFIX");
+  if (conda_prefix) {
+    std::string expected(conda_prefix);
+    // this feedstock only compiles on linux, so this test is linux-only
+    // (since we need target_platform == cross_target_platform to execute it);
+    // on windows this would be "/Library/bin/../../share/zoneinfo"
+    expected += "/lib/../share/zoneinfo";
+    VERIFY( tz_dir == expected );
+  }
```

Comment:
In other words, we wouldn't be able to test (or even display for debugging) the path that our implementation ends up using like the above, unless we export another function that wraps `zoneinfo_dir_override`. Since `zoneinfo_file` does _exactly_ that, I think it's worth considering exporting that, for several reasons:
* it's a private namespace, so no-one is supposed to rely on it anyway
* being able to debug/display the value we end up using is useful
* it doesn't change the implementation, and is trivial except the namespace-change

Reply:
@isuruf @mbargull, would you prefer:
* not moving `zoneinfo_file` into `__gnu_cxx`, at the cost of losing the ability to test our implementation explicitly
* do the move (for the reasons above), and keep the tests

Reply:
Ah, nevermind, we covered this with the `strace` tests...